### ISSUE TITLE
No verification when user deletes it's info

### ISSUE THREAD

**Issue by Unit111** [CiviWiki/OpenCiviWiki#127]
If I am a logged in user with properly entered information and I go to Settings and delete my first and last name, the changes are saved without the application telling me they are mandatory.
Then, if I decide to open another page, I am redirected to the registration form to complete my sign up.

**Comment by bradserlis** [comment]
Hello, 

we had a suggested fix: 
Within **project/api/forms.py:** 

Can this be resolved by making the first name and last name fields unable to be submit as blank, by making those fields have the property "required=True"?

class UpdateAccount(forms.ModelForm):
    """
    Form for updating Account data
    """
    class Meta:
        model = Account
        fields = ['first_name', 'last_name', 'about_me', 'profile_image']

    first_name = forms.CharField(label='First Name', max_length=63, required=**False**) **(True)**
    last_name = forms.CharField(label='Last Name', max_length=63, required=**False**) **(True)**
    about_me = forms.CharField(label='About Me', max_length=511, required=False)
    profile_image = forms.ImageField(required=False)

**Comment by khushkaushal** [comment]
Hi @brylie, I can take a look at this issue.

**Comment by brylie** [comment]
@khushkaushal, sure thanks! First steps are to

1. [install the CiviWiki development environment](https://github.com/CiviWiki/OpenCiviWiki/blob/develop/CONTRIBUTING.md#development)
2. create a superuser
3. populate some profile data for your user
4. try to replicate the bug

**Comment by khushkaushal** [comment]
@brylie I was able to reproduce the issue. I have created a PR to fix it. Requesting a review. Please and thanks!

**Comment by brylie** [comment]
Let's keep the first_name and last_name fields optional and fix the issue in code that depends on those fields.

**Comment by benwade91** [comment]
@brylie Would it be okay to default a blank first_name to 'ANONYMOUS'?

**Comment by brylie** [comment]
Sure. E.g. when rendering a Civi written by a user without first and last name, correct?

**Comment by sidanand67** [comment]
Hello @brylie,
If this issue is still not fixed then can I fix it?
Thanks.

**Comment by brylie** [comment]
Sure thing, @sidanand67. Thanks!

**Comment by mnoguier** [comment]
I would like to be assigned to this if possible?

**Comment by mnoguier** [comment]
If I understand this correctly I think the issue is coming from 
![Screen Shot 2021-12-16 at 9 08 33 PM](https://user-images.githubusercontent.com/55517552/146477149-7a066282-aabd-4acc-801d-bb26a43ae179.png)
This is checking whether or not registration is complete.  This check is done by the save function:

![Screen Shot 2021-12-16 at 9 09 47 PM](https://user-images.githubusercontent.com/55517552/146477261-c5da59cf-18de-4e1c-8fe7-394a64cb0152.png)
 and that check is done by the is full profile function which is based on whether or not they have a first and last name:

![Screen Shot 2021-12-16 at 9 10 44 PM](https://user-images.githubusercontent.com/55517552/146477324-d87724e7-3223-4145-ae09-af079db2f0c7.png)

I propose changing it to a a simple Boolean function of whether or not they finished registering

**Comment by brylie** [comment]
@mnoguier, sounds good. I've assigned you to the issue. Thanks :smiley:

**Comment by Divyashree-iyer** [comment]
If this issue is still open I would like to be assigned.
But i have some doubts that I would like to get cleared before starting -
1. what is the main purpose here? Do we simply need to keep the first name and last name mandatory? if yes then it has already been solved by @khushkaushal 
2. if you want to keep the first name and last name as optional but give a default value as "ANONYMOUS" 
3.  or change the is_full_profile function in /project/accounts/models.py to not consider first name and last name as an important indicator of the profile being completed, as suggested by @mnoguier 
4. make changes in such a way that even if there is no first_name and last_name field code must not break
5. Or do you want something totally different?

**Comment by shashwatpandeyvns** [comment]
Hi,
I would like contribute to the project. How can start this journey? Please, Enlighten me.

**Comment by werberger** [comment]
Looking for first contribution. Relatively new to Python but this looks achievable. Is it being worked on by anyone? Please assign me the issue if not.

Any idea why it keeps getting started and seemingly fixed but nothing merged?

**Comment by quinnrdavis** [comment]
I'd like to take this on. Can I ask for some clarification on what the desired functionality is? Do we want the user to not be able to delete their first and last name? Or make first and last name not mandatory?

**Comment by brylie** [comment]
@quinnrdavis, thanks! I'll assign you.

**Comment by quinnrdavis** [comment]
Thanks! I am able to reproduce the bug. I notice that the registration page does not let you continue without something in both the first and last name fields so I assume we don't want any users with a blank first or last name? It seems at odds with the comments above about defaulting a blank name to ANONYMOUS so I'd just like to understand what precisely our goal is.

**Comment by brylie** [comment]
Ah, right. This issue will be resolved with PR #1428

**Comment by quinnrdavis** [comment]
I can't really work on this without knowing what the goal is so I'm gonna go ahead and unassign myself for now. If you know the direction you want to go you can reassign me and I'll be happy to try and figure it out.

**Comment by brylie** [comment]
I believe the name fields should be made optional, and the templates should show the username when the name fields are not available.

**Comment by quinnrdavis** [comment]
Great! That sounds very doable. If you would like to assign me again I can work on that.

**Comment by brylie** [comment]
If it's not too much of a difficulty. Let's move the `first_name` and `last_name` fields to the `Profile` model instead of `Account`. The fields will need to be set to `None` on the `Account` model.

```py
class Accounts(AbstractUser):
    first_name = None
    last_name = None
    ...
```

**Comment by werberger** [comment]
I have finally got my branch set up. However I am unable to reproduce the issue. When I register it only asks for Username, Email, and Password....?

**Comment by brylie** [comment]
Yes, we've removed the account setup "wizard" so users have a more straightforward onboarding process. However, users can edit their profile to fill in more details by clicking the profile link in the navigation menu.

**Comment by werberger** [comment]
I should have said that I can edit a logged in user from the 'Settings' menu item and add or delete the first and/or last names and it does not throw any error, nor ask me to complete registration.

Perhaps I'm misunderstanding the problem, or have not installed it properly (as I also cannot add issues of interest, or create new threads).

**Comment by brylie** [comment]
OK, well, it seems this may be a non-issue. This old bug came out of our QA process and may have been indirectly resolved along the way. Thanks for investigating, @werberger!

**Comment by brylie** [comment]
@all-contributors please add @werberger for bug, qa

**Comment by werberger** [comment]
I may have commented too soon ◞‸◟

I was playing around in the admin console and think I've found it still exists.

That is, if you try and save a _Profile_ with empty _First name_ or _Last name_.

You are able to save a _User_ with name fields empty, just not a _Profile_.

Do you want to reopen this or create a new issue?

**Comment by brylie** [comment]
Let's open a new issue so we can start the discussion fresh.

**Comment by werberger** [comment]
sorry, did you want me to open the new issue @brylie?

**Comment by brylie** [comment]
Yeah. Open the issue, and we will discuss the ideal solution.
